twitter4j-core/src/internal-http/java/twitter4j/HttpClientImpl.java

```diff
@@ -83,6 +88,20 @@ public HttpResponse post(String url, HttpParameter[] params) throws TwitterExcep
         return request(new HttpRequest(RequestMethod.POST, url, params, null, null));
     }
 
+    private SSLContext getTlsSslContext() throws TwitterException {
+        // https://stackoverflow.com/questions/30121510/java-httpsurlconnection-and-tls-1-2
+        // If the KeyManager[] parameter is null, the installed security providers are searched for the highest-priority implementation of the KeyManagerFactory, from which an appropriate KeyManager is obtained.
+        // If the TrustManager[] parameter is null, the installed security providers are searched for the highest-priority implementation of the TrustManagerFactory, from which an appropriate TrustManager is obtained.
+        // Likewise, the SecureRandom parameter can be null, in which case a default implementation is used.
+        try {
+            SSLContext sc = SSLContext.getInstance("TLSv1.2");
+            sc.init(null, null, new java.security.SecureRandom());
+            return sc;
+        } catch (KeyManagementException | NoSuchAlgorithmException e) {
+            throw new TwitterException("Generating TLSv1.2 SSLContext failed", e);
+        }
+    }
```

Comment:
It seems like this will generate a new `SSLContext` for each request, should we generate a single one in the client constructor and store it in a field?

Reply:
Great point. I'll update to do that.